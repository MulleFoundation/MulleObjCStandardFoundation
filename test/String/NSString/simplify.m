//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//

#ifdef __MULLE_OBJC__
# import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>
#else
# import <Foundation/Foundation.h>
#endif


static void  test( NSString *s)
{
   NSString   *simplified;

   simplified = [s mulleStringBySimplifyingComponentsSeparatedByString:@"/"
                                                          simplifyDots:YES];
   printf( "\"%s\" simplifies to \"%s\"\n",
            [s UTF8String],
            [simplified UTF8String]);
}

int main( int argc, const char * argv[])
{
   test( @"");

   // single char
   test( @"/");
   test( @".");
   test( @"..");
   test( @"a");

   // / prefix
   test( @"//");
   test( @"/.");
   test( @"/..");
   test( @"/a");

   // suffix
   test( @"./");
   test( @"../");
   test( @"a/");

   // infix
   test( @"///");
   test( @"/./");
   test( @"/../");
   test( @"/a/");

   // combinations
   test( @"/a/.");
   test( @"/./a");
   test( @"/a/./");
   test( @"/./a/");

   test( @"a/.");
   test( @"./a");
   test( @"a/./");
   test( @"./a/");

   test( @"/a/..");
   test( @"/../a");
   test( @"/a/../");
   test( @"/../a/");

   test( @"a/..");
   test( @"../a");
   test( @"a/../");
   test( @"../a/");

   test( @"a/../b");
   test( @"a/./b");
   test( @"a/../b/");
   test( @"a/./b/");
   test( @"/a/../b");
   test( @"/a/./b");
   test( @"/a/../b/");
   test( @"/a/./b/");

   test( @"/Volumes/Source/srcO/MulleFoundation/MulleObjCOSFoundation/test/NSBundle");

   test( @"../..");
   test( @"a/../..");
   test( @"/a/../..");

   return( 0);
}
