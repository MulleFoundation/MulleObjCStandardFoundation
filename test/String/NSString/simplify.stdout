"" simplifies to ""
"/" simplifies to "/"
"." simplifies to "."
".." simplifies to ".."
"a" simplifies to "a"
"//" simplifies to "/"
"/." simplifies to "/"
"/.." simplifies to "/"
"/a" simplifies to "/a"
"./" simplifies to "./"
"../" simplifies to "../"
"a/" simplifies to "a/"
"///" simplifies to "/"
"/./" simplifies to "/"
"/../" simplifies to "/"
"/a/" simplifies to "/a/"
"/a/." simplifies to "/a"
"/./a" simplifies to "/a"
"/a/./" simplifies to "/a/"
"/./a/" simplifies to "/a/"
"a/." simplifies to "a"
"./a" simplifies to "./a"
"a/./" simplifies to "a/"
"./a/" simplifies to "./a/"
"/a/.." simplifies to "/"
"/../a" simplifies to "/a"
"/a/../" simplifies to "/"
"/../a/" simplifies to "/a/"
"a/.." simplifies to "."
"../a" simplifies to "../a"
"a/../" simplifies to "./"
"../a/" simplifies to "../a/"
"a/../b" simplifies to "./b"
"a/./b" simplifies to "a/b"
"a/../b/" simplifies to "./b/"
"a/./b/" simplifies to "a/b/"
"/a/../b" simplifies to "/b"
"/a/./b" simplifies to "/a/b"
"/a/../b/" simplifies to "/b/"
"/a/./b/" simplifies to "/a/b/"
"/Volumes/Source/srcO/MulleFoundation/MulleObjCOSFoundation/test/NSBundle" simplifies to "/Volumes/Source/srcO/MulleFoundation/MulleObjCOSFoundation/test/NSBundle"
"../.." simplifies to "../.."
"a/../.." simplifies to ".."
"/a/../.." simplifies to "/"
